dbt/include/bigquery/macros/materializations/incremental_strategy/insert_overwrite.sql

```diff
@@ -19,7 +19,9 @@
 {% macro bq_copy_partitions(tmp_relation, target_relation, partitions, partition_by) %}
 
   {% for partition in partitions %}
-    {% if partition_by.granularity == 'hour' %}
+    {% if partition_by.data_type == 'int64' %}
+      {% set partition = partition | as_text %}
```

Comment:
For the record, this is fixing copy_partitions setting for `data_type: "int64"` (which now tested)